### PR TITLE
Update the parameters description in the summary as described in the issue 2177

### LINKED ISSUE #2177: The parameter descriptions should distinguish between columns and column names.
As @wschin pointed out here: https://github.com/dotnet/machinelearning/pull/2170#issuecomment-455346083 the parmaters, and in general our XML documentation should distinguish between columns and column names. 

This issue will be considered closed when all the params of this language:

```
/// <param name="labelColumn">The label column.</param>
/// <param name="featureColumn">The featureColumn column.</param>
/// <param name="weights">The optional weights column.</param>
```
Get changed to: 
```
/// <param name="labelColumnName">The name of the label column.</param>
/// <param name="featureColumnName">The name of the feature column.</param>
/// <param name="weightsColumnName">The name of the optional weights column.</param>
```

### PR DESCRIPTION
**Pull request by HerraHak**
Fixes #2177 

The description of the parameters have been updated as described by @sfilipi in issue #2177 

We are excited to review your PR.

So we can do the best job, please check:

- [X ] There's a descriptive title that will make sense to other developers some time from now. 
- [X ] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [X ] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [ N/A] You have included any necessary tests in the same PR.

### CODE REVIEW

**Comment by HerraHak** [comment]
to be honest I don't know why the code coverage failed...  Is there an issue with master?

**Comment by artidoro** [comment]
We have sporadic failures due to the random nature of our tests, I just restarted the code coverage bot. Thanks for your PR!

**Comment by Ivanidzo4ka** [comment]
@sfilipi  do you mean in issue to unify all column or only specific one?
Because I can see following variation in code for public classes:
` /// <param name="labelColumn">The label column name.</param>`
`/// <param name="labelColumn">The label, or dependent variable.</param>`
`/// <param name="labelColumn">The labelColumn, or dependent variable.</param>`
`/// <param name="labelColumn">Name of the column to use for labels.</param>`
`/// <param name="labelColumn">The labelColumn column.</param>`

**Comment by sfilipi** [comment]
@HerraHak thanks for your PR. 

I have edited the original issue to describe other feedback that we have had. 

this is how those parameters should look like:

/// <param name="labelColumnName">The name of the label column.</param>
/// <param name="featureColumnName">The name of the feature column.</param>
/// <param name="weightsColumnName">The name of the optional weights column.</param>

There are many of them, through the code-base. If you don't intend to fix all of them, please don;t close the issue. 

@Ivanidzo4ka does the above look good? as long as we standardize to something. 
Feel free to propose different language.

File: src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs [hunk 1]
```diff
@@ -17,9 +17,9 @@ public static class TreeExtensions
         /// Predict a target using a decision tree regression model trained with the <see cref="FastTreeRegressionTrainer"/>.
         /// </summary>
         /// <param name="catalog">The <see cref="RegressionCatalog"/>.</param>
-        /// <param name="labelColumn">The label column.</param>
-        /// <param name="featureColumn">The feature column.</param>
-        /// <param name="weights">The optional weights column.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weights">The name of the optional weights column.</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs]: so weightsColumnName, labelColumnName, featureColumnName.

**Comment by Ivanidzo4ka** [comment]
I'm more than happy with proposed language. I just wonder what is criteria to close issue :)

---
In reply to: [461206771](https://github.com/dotnet/machinelearning/pull/2432#issuecomment-461206771) [](ancestors = 461206771)

**Comment by HerraHak** [comment]
So if I understand correctly, the language like it is in the pull request is what we want?

If that's the case, I can look for other occurrences in the code and change that.

Hakim BEN AMAR

________________________________

File: src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs (hunk 1 above)
Comment (HerraHak) [reply src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs]: Should we change the name of the parameters too or is a different issue?

**Comment by HerraHak** [comment]
Huh, the code coverage failed again.
Could anybody perhaps relaunch the code coverage test??

File: src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs]: If it's not hard for you.
This weights is actually `weightsColumn`
or `weightColumn`

@sfilipi why we use singular for feature column and plural for weight column?

---
In reply to: [254624678](https://github.com/dotnet/machinelearning/pull/2432#discussion_r254624678) [](ancestors = 254624678)

File: src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs [hunk 2]
```diff
@@ -20,9 +20,9 @@ public static class HalLearnersCatalog
         /// Predict a target using a linear regression model trained with the <see cref="OlsLinearRegressionTrainer"/>.
         /// </summary>
         /// <param name="catalog">The <see cref="RegressionCatalog"/>.</param>
-        /// <param name="labelColumn">The labelColumn column.</param>
-        /// <param name="featureColumn">The features column.</param>
-        /// <param name="weights">The weights column.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the features column.</param>
+        /// <param name="weights">The name of the weights column.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs]: >weights [](start = 50, length = 7)

optional

File: src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs [hunk 3]
```diff
@@ -89,8 +89,8 @@ public static ImagePixelExtractingEstimator ExtractPixels(this TransformsCatalog
         /// <seealso cref= "ImageLoadingEstimator" />
         /// </remarks >
         /// <param name="catalog">The transform's catalog.</param>
-        /// <param name="inputColumnName">Name of the input column.</param>
-        /// <param name="outputColumnName">Name of the resulting output column.</param>
+        /// <param name="inputColumnName">The name of the input column.</param>
+        /// <param name="outputColumnName">The name of the resulting output column.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs]: I would prefer standardize language we use for them:
```
/// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
/// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
        
```

File: src/Microsoft.ML.ImageAnalytics/ImagePixelExtractor.cs [hunk 4]
```diff
@@ -700,7 +700,7 @@ internal void Save(ModelSaveContext ctx)
         ///</summary>
         /// <param name="env">The host environment.</param>
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>. Null means <paramref name="inputColumnName"/> is replaced.</param>
-        /// <param name="inputColumnName">Name of the input column.</param>
+        /// <param name="inputColumnName">The name of the input column.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.ImageAnalytics/ImagePixelExtractor.cs]: > /// <param name="inputColumnName">The name of the input column.</param> [](start = 7, length = 72)

```
/// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
/// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
```

File: src/Microsoft.ML.ImageAnalytics/ImageResizer.cs [hunk 5]
```diff
@@ -135,7 +135,7 @@ private static VersionInfo GetVersionInfo()
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="imageWidth">Width of resized image.</param>
         /// <param name="imageHeight">Height of resized image.</param>
-        /// <param name="inputColumnName">Name of the input column.</param>
+        /// <param name="inputColumnName">The name of the input column.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.ImageAnalytics/ImageResizer.cs]: >inputColumnName [](start = 25, length = 15)

 /// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>

File: src/Microsoft.ML.LightGBM/LightGbmCatalog.cs [hunk 6]
```diff
@@ -55,9 +55,9 @@ public static class LightGbmExtensions
         /// Predict a target using a decision tree binary classification model trained with the <see cref="LightGbmBinaryTrainer"/>.
         /// </summary>
         /// <param name="catalog">The <see cref="BinaryClassificationCatalog"/>.</param>
-        /// <param name="labelColumn">The labelColumn column.</param>
-        /// <param name="featureColumn">The features column.</param>
-        /// <param name="weights">The weights column.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the features column.</param>
+        /// <param name="weights">The name of the weights column.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.LightGBM/LightGbmCatalog.cs]: >weights [](start = 50, length = 7)

optional

File: src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs [hunk 7]
```diff
@@ -45,7 +45,7 @@ internal LightGbmMulticlassTrainer(IHostEnvironment env, Options options)
         /// Initializes a new instance of <see cref="LightGbmMulticlassTrainer"/>
         /// </summary>
         /// <param name="env">The private instance of <see cref="IHostEnvironment"/>.</param>
-        /// <param name="labelColumn">The name of The label column.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
         /// <param name="featureColumn">The name of the feature column.</param>
         /// <param name="weights">The name for the column containing the initial weight.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs]: >The name for the column containing the initial weight. [](start = 34, length = 54)

`/// <param name="weightsColumnName">The name of the optional weights column.</param>`

**Comment by Ivanidzo4ka** [comment]
>         /// <param name="weightsColumn">The name of the weights column.</param>

`/// <param name="weightsColumnName">The name of the optional weights column.</param>`

---
Refers to: src/Microsoft.ML.StandardLearners/Standard/Online/OnlineGradientDescent.cs:102 in 6bf8a17. [](commit_id = 6bf8a1780e5801ec9fad2145bc3ed67f8b21b6ed, deletion_comment = False)

File: src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs (hunk 3 above)
Comment (HerraHak) [reply src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs]: Added the changes in the last update of the PR

File: src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs [hunk 8]
```diff
@@ -17,25 +17,25 @@ public static class TreeExtensions
...
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs]: >weights [](start = 25, length = 7)

can you update this one as well, otherwise build is not working

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs]: Sorry I meant in whole file. We have strict rules to match parameters names in xml documentation

---
In reply to: [255319474](https://github.com/dotnet/machinelearning/pull/2432#discussion_r255319474) [](ancestors = 255319474)

Comment (HerraHak) [reply src/Microsoft.ML.FastTree/TreeTrainersCatalog.cs]: Should be ok now

File: src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs [hunk 9]
```diff
@@ -89,8 +89,8 @@ public static ImagePixelExtractingEstimator ExtractPixels(this TransformsCatalog
         /// <seealso cref= "ImageLoadingEstimator" />
         /// </remarks >
         /// <param name="catalog">The transform's catalog.</param>
-        /// <param name="inputColumnName">Name of the input column.</param>
-        /// <param name="outputColumnName">Name of the resulting output column.</param>
+        /// <param name="inputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
+        /// <param name="outputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs]: >outputColumnName [](start = 25, length = 16)

you need to switch input and output column. Output column is resulting and input column is name of column to transform.

File: src/Microsoft.ML.LightGBM/LightGbmCatalog.cs (hunk 6 above)
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.LightGBM/LightGbmCatalog.cs]: >the [](start = 46, length = 3)

optional weights column

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.LightGBM/LightGbmCatalog.cs]: across whole file

---
In reply to: [255809893](https://github.com/dotnet/machinelearning/pull/2432#discussion_r255809893) [](ancestors = 255809893)

File: src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs (hunk 9 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs]: This is how it suppose to be.
```
 /// <param name="outputColumnName"> Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
/// <param name="inputColumnName"> Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
```
And now check what do you have in `param name=`

---
In reply to: [255809779](https://github.com/dotnet/machinelearning/pull/2432#discussion_r255809779) [](ancestors = 255809779)

**Comment by Ivanidzo4ka** [comment]
@abgoswam covered same thing in his PR.
Thank you for help!

**Comment by HerraHak** [comment]
Well it was nice trying to contribute on this.  I'll pick up another issue then.
